iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber_uds.inl

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_UDS_INL
+
+#include "owl/kom/event_subscriber_uds.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventSubscriber<T, EventTransmission::UDS>::EventSubscriber(const ServiceIdentifier&,
+                                                                   const InstanceIdentifier& instance,
+                                                                   const EventIdentifier&) noexcept
+    : m_uds(iox::posix::UnixDomainSocket::create(instance, iox::posix::IpcChannelSide::SERVER)
+                .expect("Failed to create UNIX domain socket!"))
+{
+}
+
+template <typename T>
+inline EventSubscriber<T, EventTransmission::UDS>::~EventSubscriber() noexcept
+{
+    m_run = false;
+    m_thread.join();
+}
+
+template <typename T>
+inline void EventSubscriber<T, EventTransmission::UDS>::Subscribe(std::size_t) noexcept
+{
+    // Subscribe not supported with UDS
+}
+
+template <typename T>
+void EventSubscriber<T, EventTransmission::UDS>::Unsubscribe() noexcept
+{
+    // Unsubscribe not supported with UDS
+}
+
+template <typename T>
+template <typename Callable>
+core::Result<size_t> inline EventSubscriber<T, EventTransmission::UDS>::GetNewSamples(Callable&& callable,
+                                                                                      size_t maxNumberOfSamples)
+{
+    IOX_DISCARD_RESULT(maxNumberOfSamples);
+
+    core::Result<size_t> numberOfSamples{1};
+
+    auto samplePtr = std::make_unique<SampleType>();
+    SampleType& sample = *samplePtr;
+
+    std::string tempBuffer;
+
+    // Receive the first (up to) 4095 Bytes
```

Comment:
How did you come up with 4095. This can be different on every OS, architecture or even in between linux distributions. You have to read out `IOX_UDS_SOCKET_MAX_MESSAGE_SIZE` in `platform_settings.hpp`

Reply:
Closing, as this file will be moved to a separate repo.